src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Compilation.cs

```diff
@@ -202,6 +202,11 @@ public MethodDesc ExpandIntrinsicForCallsite(MethodDesc intrinsicMethod, MethodD
             }
             else if (intrinsicOwningType.Name.SequenceEqual("MethodBase"u8) && intrinsicOwningType.Namespace.SequenceEqual("System.Reflection"u8))
             {
+                if (callsiteMethod.IsAsyncVariant())
```

Comment:
Nit: we should do this inside the below if block because this swap is specific to `GetCurrentMethod`, not all methods on `MethodBase`.